src/library/scala/collection/IterableOnce.scala

```diff
@@ -458,6 +458,26 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
     */
   def span(p: A => Boolean): (C, C)
 
+  /** Splits this $coll into a prefix/suffix pair at a given position.
+   *
+   *  Note: `c splitAt n` is equivalent to (but possibly more efficient than)
+   *         `(c take n, c drop n)`.
+   *  $orderDependent
+   *
+   *  @param n the position at which to split.
+   *  @return  a pair of ${coll}s consisting of the first `n`
+   *           elements of this $coll, and the other elements.
+   *  @note    Reuse: $consumesOneAndProducesTwoIterators
+   */
+  def splitAt(n: Int): (C, C) = {
+    var i = 0
+    this span { _ =>
+      val res = n < i
```

Comment:
I think you mean `i < n`.  Then it will work fine.  No test is necessary; collections-laws would catch this.

Reply:
Also, I think the `if (i < n) { i += 1; true } else false` way of expressing the logic is clearer.

Reply:
good catch!

Reply:
@Ichoran I don't think collections-laws would catch this (at least without being updated), as this method did not previously exist on `IterableOnce`